src/Microsoft.Extensions.FileSystemGlobbing/Abstractions/InMemoryDirectoryInfo.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Extensions.FileSystemGlobbing.Abstractions
```

Comment:
Why is this in abstractions and not a private implementation of the globbing package?